hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/kafka/PropertiesResolver.java

```diff
@@ -121,8 +116,15 @@ private static void withSerdeConsumerProperties(
         String format = options.get(isKey ? OPTION_KEY_FORMAT : OPTION_VALUE_FORMAT);
         if (format == null && isKey) {
             properties.putIfAbsent(deserializer, BYTE_ARRAY_DESERIALIZER);
-        } else if (JAVA_FORMAT.equals(format)) {
-            String clazz = options.get(isKey ? SqlConnector.OPTION_KEY_CLASS : SqlConnector.OPTION_VALUE_CLASS);
+        } else if (AVRO_FORMAT.equals(format)) {
+            properties.putIfAbsent(deserializer, AVRO_DESERIALIZER);
+        } else if (JSON_FORMAT.equals(format)) {
+            properties.putIfAbsent(deserializer, BYTE_ARRAY_DESERIALIZER);
+        } else {
+            String clazz = JavaClassNameResolver.resolveClassName(
+                    format,
+                    options.get(isKey ? SqlConnector.OPTION_KEY_CLASS : SqlConnector.OPTION_VALUE_CLASS)
+            );
```

Comment:
I don't like this. So if there's `keyJavaClass` in the options, we ignore the format? If one does
```
OPTIONS (
    'keyFormat'='bigint',
    'keyJavaClass'='java.lang.Integer'
)
```
It will work exactly as `'keyFormat'='int'`. I think we should always check the format and if, based on that format, some other property is needed, use that. That is, if `keyFormat` is `java`, look ak `keyJavaClass`. If `keyFormat` is `int`, don't look at any other property (as we ignore unneeded properties).

Same below in `withSerdeProducerProperties`.